src/main/java/com/empowerops/linqalike/IList.java

```diff
@@ -0,0 +1,167 @@
+    }
+
+    @Override
+    public IList<T> with(T toInclude) {
+        return new IList<>(backingList.plus(toInclude));
+    }
+
+    public IList<T> with(int index, T toInclude){
+        //TODO hmm, this means my 'with' is their 'add' and their 'with' is juc's 'set' method.
+        return new IList<>(backingList.plus(index, toInclude));
+    }
+
+    @Override
+    public IList<T> with(T another0, T another1) {
+        return new IList<>(backingList.plusAll(Arrays.asList(another0, another1)));
+    }
+
+    @Override
+    public IList<T> with(T another0, T another1, T another2) {
+        return new IList<>(backingList.plusAll(Arrays.asList(another0, another1, another2)));
+    }
+
+    @Override
+    public IList<T> with(T another0, T another1, T another2, T another3) {
+        return new IList<>(backingList.plusAll(Arrays.asList(another0, another1, another2, another3)));
+    }
+
+    @Override
+    public IList<T> with(T another0, T another1, T another2, T another3, T another4) {
+        return new IList<>(backingList.plusAll(Arrays.asList(another0, another1, another2, another3, another4)));
+    }
+
+    @Override
+    @SafeVarargs
+    @SuppressWarnings("unchecked")
+    public final IList<T> with(T... othersToInclude) {
+        return new IList<>(backingList.plusAll(Arrays.asList(othersToInclude)));
+    }
+
+    @Override
+    public IList<T> with(Iterable<? extends T> toInclude) {
+        Collection<? extends T> source = toInclude instanceof Collection
+                ? (Collection) toInclude
+                : Factories.asList(toInclude);
+
+        return new IList<>(backingList.plusAll(source));
+    }
+
+    /** {@inheritDoc} **/
+    @Override
+    @Deprecated
+    public boolean add(T t) {
+        return backingList.add(t);
+    }
+
+    /**{@inheritDoc}*/ @Override @Deprecated public boolean remove(Object o) { return backingList.remove(o); }
+    /**{@inheritDoc}*/ @Override @Deprecated public boolean containsAll(Collection<?> c) {return backingList.containsAll(c);}
+    /**{@inheritDoc}*/ @Override @Deprecated public boolean addAll(Collection<? extends T> c) {return backingList.addAll(c);}
+    /**{@inheritDoc}*/ @Override @Deprecated public boolean addAll(int index, Collection<? extends T> c) {return backingList.addAll(index, c);}
+    /**{@inheritDoc}*/ @Override @Deprecated public boolean removeAll(Collection<?> c) {return backingList.removeAll(c);}
+    /**{@inheritDoc}*/ @Override @Deprecated public boolean retainAll(Collection<?> c) {return backingList.retainAll(c);}
+    /**{@inheritDoc}*/ @Override @Deprecated public void clear() {backingList.clear();}
+    /**{@inheritDoc}*/ @Override @Deprecated public T set(int index, T element) {return backingList.set(index, element);}
+    /**{@inheritDoc}*/ @Override @Deprecated public void add(int index, T element) {backingList.add(index, element);}
+    /**{@inheritDoc}*/ @Override @Deprecated public T remove(int index) {return backingList.remove(index);}
```

Comment:
TODO: in intelliJ its inheriting the lists doc, which is a pain because that doc doesn't deprecate itself (obviously), and from what I can tell I cant specify which doc to inherit.